.github/actions/check-release/action.yml

```diff
@@ -0,0 +1,63 @@
+name: Check release
+description: Check for conflicts in packages being released in this PR.
+
+inputs:
+  before:
+    description: 'The commit SHA before the current commit or pull request.'
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Check commits for changes in released packages
+      shell: bash
+      env:
+        BEFORE: ${{ inputs.before }}
+      run: |
+        set -euo pipefail
+        git fetch origin main
+
+        mapfile -t PACKAGES < <(find packages -maxdepth 2 -name "package.json" -not -path "*/node_modules/*")
+        RELEASED_PACKAGES=()
+
+        # Get all packages being released in this PR
+        for package in "${PACKAGES[@]}"; do
+          MAIN_VERSION=$(git show "refs/remotes/origin/main:$package" | jq -r .version)
```

Comment:
This comparison doesn't seem quite right. We're using `origin/main` as the "before the current commit" comparison point, but there might be more changes in-between `origin/main` and the current commit. e.g. if there is a release PR queued before the current one